### ISSUE TITLE
Automatic validity check on layers

### ISSUE THREAD

**Issue by nryeng** [awesome-raccoons/gqt#35]
An automatic check of geometry validity (the Geometry.isValid() function) for each layer. Show an icon or signal in another way which layers contain invalid geometries.

**Comment by johannbm** [comment]
Right now we have a feature that shows an alert dialog whenever you try to update a layer to an invalid WKT string. Are these two issues mutually exclusive? Will there by cases that bypass the parser without throwing an exception and is still invalid?

Right now, these mistakes will be caught and displaying an error box.
POLYGON(0 0, 0 10, 10 10, 10 0, a a)
POLYGON(0 0, 0 10, 10 10, 10 0, 1 1)

In other words, do you know of some cases that will not cast a com.vividsolutions.jts.io.ParseException when trying to parse the string, but still return false from isValid() ?

**Comment by nryeng** [comment]
Geometric invalidity is a different thing. It's about self-intersecting polygons, holes outside the exterior of the polygon, etc.

Examples:
- POLYGON((0 0, 10 0, 0 10, 10 10, 0 0))
- POLYGON((0 0, 10 0, 10 10, 0 10, 0 0),(20 20, 30 20, 30 30, 20 30, 20 20))
